documentation/api/v2g.rst

```diff
@@ -0,0 +1,139 @@
+.. _v2g:
+
+Vehicle-to-grid
+---------------
+
+As a demonstration of how to construct a suitable flex model for a given use case, we consider a client using FlexMeasures to compute :abbr:`V2G (vehicle-to-grid)` schedules. In this example, the client is interested in the following:
+
+1. :ref:`battery_protection`: Protect the battery from degradation by constraining any cycling between 25% and 85% of its available storage capacity.
+2. :ref:`car_reservations`: Ensure a minimum :abbr:`SoC (state of charge)` of 95% based on a reservation calendar for the car.
+3. :ref:`earning_by_cycling`: Use the car battery to earn money (given some dynamic tariff) so long as the above constraints are met.
+
+The following chart visualizes how constraints 1 and 2 can be formulated within a flex model, such that the resulting scheduling problem becomes feasible. A solid line shows a feasible solution, and a dashed line shows an infeasible solution.
+
+.. image:: https://github.com/FlexMeasures/screenshots/raw/main/tut/v2g_minima_maxima.png
+    :align: center
+|
+
+
+.. _battery_protection:
+
+Battery protection
+==================
+
+Let's consider a car battery with a storage capacity of 60 kWh, to be scheduled in 5-minute intervals.
+Constraining the cycling to occur within a static 25-85% SoC range can be modelled through the following ``soc-min`` and ``soc-max`` fields of the flex model:
+
+.. code-block:: json
+
+    {
+        "flex-model": {
+            "soc-min": 15,
+            "soc-max": 51,
+            "soc-unit": "kWh"
+        }
+    }
+
+A starting SoC below 15 kWh (25%) will lead to immediate charging to get within limits (as shown above).
+Likewise, a starting SoC above 51 kWh (85%) would lead to immediate discharging.
+Setting a SoC target outside of the static range leads to an infeasible problem and will be rejected by the FlexMeasures API.
+
+To enable a target SoC above 85% (as shown above), it is necessary to set the ``soc-max`` field to 60 kWh (100%), and to instead use the ``soc-maxima`` field to convey the desired upper limit for regular cycling:
+
+.. code-block:: json
+
+    {
+        "flex-model": {
+            "soc-min": 15,
+            "soc-max": 60,
+            "soc-maxima": [
```

Comment:
Side-question: Wouldn't it be nice to allow to specify ``sox-maxima`` (and minima of course) as a range, so one would not have to send all consecutive points?

```
"soc-maxima": {
    "value": 51,
    "start": datetime": "2024-02-05T12:00:00+01:00",
    "end": datetime": "2024-02-05T13:30:00+01:00"
}
```

One problem I see is overlaps but that should cause an error.

Reply:
I agree such a syntax would be a lot less verbose. Worth exploring as a follow-up.

Reply:
https://github.com/FlexMeasures/flexmeasures/issues/898